### PR TITLE
Upgraded seed creating to handle argument passing and perform validation

### PR DESCRIPTION
**Pull request by danyoungday**
This is a little yucky at the moment because the prescriptors aren't set up with the idea of using them to evaluate candidates separate from their save structure. Would be worth taking a look at later to not have to call any "private" methods. There are also some "hard-coded" args which can be added in later as well.

### CODE REVIEW

File: use_cases/eluc/prescriptors/esp/create_seeds.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    
+    test_df = dataset.test_df.sample(frac=0.01, random_state=100)
+    context_df = test_df[constants.CAO_MAPPING["context"]]
+
+    for seed_path in seed_dir.iterdir():
+        candidate = load_model(seed_path)
+        encoded_context_df = dataset.encoder.encode_as_df(context_df)
+        reco_land_use = dummy_prescriptor.prescribe(candidate, encoded_context_df)
+        reco_df = pd.DataFrame(reco_land_use["reco_land_use"].tolist(), columns=constants.RECO_COLS)
+        context_actions_df = dummy_prescriptor._reco_to_context_actions(reco_df, encoded_context_df)
```
Comment (danyoungday) [inline use_cases/eluc/prescriptors/esp/create_seeds.py:96]: We call _reco_to_context_actions here because there is currently no way to call prescribe_and_predict without the underlying assumption that the file structure will be the same as how the results are saved during training (as opposed to seed training)

File: use_cases/eluc/prescriptors/esp/create_seeds.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    land_use = df[constants.RECO_COLS].sum(axis=1)
+    max_change_preds = df[constants.RECO_COLS].copy()
+    max_change_preds[constants.RECO_COLS] = 0
+    max_change_preds[best_col] = land_use
+
+    y_train = max_change_preds.to_numpy()
+    X_train = [encoded_df[col].values for col in constants.CAO_MAPPING["context"]]
+    
+    max_change_model = create_template_model()
+    opt = tf.keras.optimizers.legacy.Adam(learning_rate=0.001)
```
Comment (danyoungday) [inline use_cases/eluc/prescriptors/esp/create_seeds.py:65]: Could still add more args for things like learning rate, etc.

File: use_cases/eluc/prescriptors/nsga2/create_seeds.py [hunk 3]
```diff
@@ -11,23 +12,23 @@
...
+from prescriptors.nsga2.torch_prescriptor import TorchPrescriptor
 
-def supervised_backprop(save_path: Path, ds: TorchDataset):
+def supervised_backprop(save_path: Path, ds: TorchDataset, candidate_params: dict, n_epochs=300):
     """
     Performs supervised backpropagation on the given dataset to create a Candidate.
     """
     train_ds, val_ds = random_split(ds, [int(len(ds) * 0.8), len(ds) - int(len(ds) * 0.8)])
     train_dl = DataLoader(train_ds, batch_size=4096, shuffle=True)
     val_dl = DataLoader(val_ds, batch_size=4096, shuffle=True)
```
Comment (danyoungday) [inline use_cases/eluc/prescriptors/nsga2/create_seeds.py:25]: Torch seed creation uses a validation set to track MAE, not entirely sure how to do this neatly in TensorFlow so the ESP seed creation does not.
